packages/lesswrong/components/books/Book2020FrontpageWidget.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { useUpdate } from '../../lib/crud/withUpdate';
+import { Link } from '../../lib/reactRouterWrapper';
+import { Components, registerComponent } from '../../lib/vulcan-lib';
+import { useDialog } from '../common/withDialog';
+import { useCurrentUser } from '../common/withUser';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    marginBottom: 50,
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    position: "relative",
+    width: "100%",
+    zIndex: theme.zIndexes.frontpageBooks,
+    [theme.breakpoints.down('md')]: {
+      width: '100%',
+      maxWidth: 765,
+      overflow: 'unset',
+    },
+    [theme.breakpoints.down('lg')]: {
+      left: 0,
+    },
+    [theme.breakpoints.down('xs')]: {
+      marginTop: 20,
+      marginBottom: 0
+    }
+  },
+  heading: {
+    [theme.breakpoints.down('xs')]: {
+      fontSize: '2.2rem !important',
+      marginTop: '70px !important'
+    }
+  },
+  secondaryHeading: {
+    marginTop: '-16px',
+    fontStyle: 'italic',
+    fontWeight: 'normal',
+  },
+  bookExplanation: {
+    float: 'right',
+    textAlign: "right",
+    [theme.breakpoints.down('md')]: {
+      paddingRight: 0
+    },
+    [theme.breakpoints.down('xs')]: {
+      // paddingRight: 16,
+      // textAlign: 'right',
+    }
+  },
+  learnMore: {
+    ...theme.typography.commentStyle,
+    display: 'flex',
+    alignItems: 'center',
+    height: 36,
+    fontSize: '1.2rem',
+    marginLeft: 16,
+    marginRight: 16,
+    whiteSpace: "nowrap",
+    marginTop: -7
+  },
+  buttonRow: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'flex-end',
+  },
+  closeButton: {
+    ...theme.typography.commentStyle,
+    position: 'absolute',
+    right: 0,
+    top: -8,
+    fontSize: '1rem',
+    color: theme.palette.icon.dim2,
+    cursor: 'pointer',
+  },
+  descriptionText: {
+    fontSize: '0.96em',
+    [theme.breakpoints.down('xs')]: {
+      display: "none",
+    },
+  },
+  sectionTwo: {
+    [theme.breakpoints.down('sm')]: {
+      display: "none"
+    }
+  }
+})
+
+const Book2020FrontpageWidget = ({ classes }: {
+  classes: ClassesType,
+}) => {
+  const { BookCheckout, Book2020Animation, ContentStyles, Row } = Components
+  const currentUser = useCurrentUser();
+  const { mutate: updateUser } = useUpdate({
+    collectionName: "Users",
+    fragmentName: 'UsersCurrent',
+  });
+  const { openDialog } = useDialog();
+
+  if (currentUser?.hideFrontpageBook2020Ad) return null
+
+  const hideClickHandler = async () => {
+    if (currentUser) {
+      await updateUser({
+        selector: { _id: currentUser._id },
+        data: {
+          hideFrontpageBook2020Ad: true
+        },
+      })
+    } else {
+      openDialog({
+        componentName: "LoginPopup",
+        componentProps: {}
+      });
+    }
+  }
+
+  const BookMarketingText = ({title, subtitle, description, buttons}: {
```

Comment:
Feels kinda weird for this to be defined inside another component, maybe just pull it out?  Can keep it in the same file